### PR TITLE
[17.0][MIG] agreement: Migration to 17.0 

### PR DESCRIPTION
**Pull request by ramiadavid**

### CODE REVIEW

File: agreement/models/agreement.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            "This agreement code already exists for this partner!",
+        )
+    ]
+
+    def copy(self, default=None):
+        """Always assign a value for code because is required"""
+        default = dict(default or {})
+        if default.get("code", False):
+            return super().copy(default)
+        default.setdefault("code", _("%s (copy)") % (self.code))
```
Comment (rousseldenis) [inline agreement/models/agreement.py]: Use translation function arguments instead

```suggestion
        default.setdefault("code", _("%(code)s (copy)", code=self.code))
```

**Review by rousseldenis — APPROVED** [review]
Minor comment

**Review by etobella — APPROVED** [review]
LGTM

Just waiting for the change suggested @ramiadavid

**Review by ivantodorovich — APPROVED** [review]

**Pull request merged**